sys/net/gnrc/network_layer/ipv6/nib/_nib-6ln.c

```diff
@@ -35,7 +35,7 @@ static bool _is_iface_eui64(gnrc_netif_t *netif, const eui64_t *eui64)
     eui64_t iface_eui64;
     int res = gnrc_netif_get_eui64(netif, &iface_eui64);
     return (res == sizeof(eui64_t)) &&
-           (iface_eui64.uint64.u64 == eui64->uint64.u64);
+           !memcmp(&iface_eui64, eui64, sizeof(eui64_t));
```

Comment:
... and replacing real comparisons with the less This change alone, justifies safer `memcmp()` here and in other places, justifies keeping at least an `uint64_t` version in the union to me.

Reply:
In this case I think the [eui64_compare](https://github.com/RIOT-OS/RIOT/pull/12713#issuecomment-559101676) function proposed by Kaspar could improve the readability here without the need to add an `uint64_t` variant. In other unrelated places of the RIOT codebase this is done already (e.g: `gpio_is_equal`) for custom types. And if it's ever needed to use an `uint64_t` the function could still have a place with the bonus of not changing a single line of code when that happens.